### ISSUE TITLE
Our cache handling could be improved

### ISSUE THREAD

**Issue by skx** [lumail/lumail2#227]
At the moment we cache several things:

* The formatted/rendered display of each message.
* The formatted/rendered display of each maildir.
* The ctime of each message
   * Used for `index.limit("today")` which I __love__ :)
   * Used for sorting.

If you're like me though you'll alternate between using the released version of lumail, and the git-version.  The cache is associated with a particular release and this is done by having a prefix:

     VERSION=foo
     cache_key1=cached_value1
     cache_key2=cached_value2

On startup if the current version of lumail doesn't match that in the cache-file it is blown away.  This means it is rebuilt (slowly) as you open (huge) mailboxes, etc.

Instead of using a cache __file__ we should use a cache __directory__.  That would allow us to keep multiple versions around:

     ~/.lumail2/cache/2.7
     ~/.lumail2/cache/64fd
     ~/.lumail2/cache/abd3

In short instead of having `VERSON=$version` header we could just have a per-version cache directory, writing to `~/.lumail2/cache/$VERSION`.  That would allow jumping between releases without blowing away the cache.
